scipy/signal/_peak_finding_utils.pyx

```diff
@@ -74,3 +74,70 @@ def _argmaxima1d(np.float64_t[:] x not None):
 
     maxima.resize(m, refcheck=False)  # Keep only valid part of array memory.
     return maxima
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _select_by_peak_distance(np.intp_t[:] peaks not None,
+                             np.float64_t[:] priority not None,
+                             dmin):
+    """
+    Evaluate which peaks fulfill the distance condition.
+
+    Parameters
+    ----------
+    peaks : ndarray
+        Indices of peaks in `vector`.
+    priority : ndarray
+        An array matching `peaks` used to determine priority of each peak. A peak
+        with a higher priority value is kept over one with a lower one.
+    dmin : number
+        Minimal distance that peaks must be spaced.
+
+    Returns
+    -------
+    keep : ndarray[bool]
+        A boolean mask evaluating to true where `peaks` fulfill the distance
+        condition.
+
+    Notes
+    -----
+
+    .. versionadded:: 1.1.0
+    """
+    cdef np.ndarray[np.intp_t, ndim=1] priority_to_position
+    cdef np.ndarray[np.int8_t, ndim=1] keep
+    cdef np.intp_t i, j, k, current_peak, max_index, dmin_
+
+    # Round up `dmin` because actual peak distance can only be natural number
+    dmin_ = np.intp(np.ceil(dmin))
```

Comment:
The `dmin_` variable and associated operation shown here aren't used in the profiled code in the linked notebook?

Reply:
I updated the notebook with this version of the function. The qualitative results are still the same. Sorry for the oversight on my part.